src/main.js

```diff
@@ -0,0 +1,57 @@
+import { createApp } from "vue";
+import App from "./App.vue";
+
+import "bulma/css/bulma.css";
+import "@fortawesome/fontawesome-free/css/all.css";
+import { createStore } from "vuex";
+
+const store = createStore({
+  state() {
+    return {
+      inSidebar: "",
+      postList: [],
+      currentPost: null,
+      userId: null,
+    };
+  },
+
+  mutations: {
+    addPostList(state, posts) {
+      if (Array.isArray(posts)) {
+        state.postList = [...state.postList, ...posts];
+      } else {
+        state.postList.push(posts);
+      }
```

Comment:
This mutation `addPostList` can be improved. Currently, it accepts both an array of posts and a single post. It would be better to have two separate mutations, one for adding a single post and another for adding multiple posts. This would make the code more readable and less error-prone.